### DISCUSSION TITLE
Error in Miller.m for slip system calculation (MTEX 5.5)

### DISCUSSION THREAD

**Discussion by MartinL75** [mtex-toolbox/mtex#609] [Ask Anything] [answered]
Hi,
I'm trying to run in MTEX 5.5 a previous script for slip systems in HCP. It was running OK for v. 5.4 but now I'm getting an error in the Miller classdef (mtex-5.5.0\geometry\@Miller\Miller.m) (see error mesage below).
May it have to do with my Matlab version? I'm working with 2016b.
Thanks a lot in advance.
              Martín.-

`Error using Miller`
`Error: File: Miller.m Line: 37 Column: 15`
`Unexpected MATLAB expression.`

`Error in slipSystem.basal (line 148)`
      `sS = slipSystem(Miller(1,1,-2,0,cs,'uvtw'),Miller(0,0,0,1,cs,'hkil'),varargin{:});`

`Error in CDRX_mtex550 (line 59)`
`sS_basal = slipSystem.basal(CS);`

Lines 36-38 are:
`properties
    dispStyle (1,1) MillerConvention = 'hkl' % output convention hkl or uvw
 end`

**Comment by kilir** [comment]
Hi,
I would say this is due to your matlab version and the recent changes in how `dispStyle` is handled.
Any chance to try a newer version?
Cheers,
Rüdiger

**Reply by MartinL75** [reply]
Hi Rüdiger,
Thanks for the quick response.  I don´t think I would be able to get a newer Matlab version, but anyway I'll keep going with MTEX 5.4.
Regards

**Comment by ralfHielscher** [comment]
Hi Martin,

could you please try if it works when your replace in Line: 37 of Miller.m
 (just type `edit Miller` into the command line)
the code
```
dispStyle (1,1) MillerConvention = 'hkl' % output convention hkl or uvw
```
by
```
dispStyle = MillerConvention.hkl
```
Thank you very much,
Ralf.

********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by MartinL75** [reply]
Hi Ralf:
Thank you very much for the suggestion. I tried it and it worked... However, I got a new error coming from another line below the previous one:

Error using `sum`
Dimension argument must be a positive integer scalar within indexing range.
Error in `tensor/EinsteinSum` (line 86)
`if rDel>0, M1 = sum(M1,1:rDel); end`
Error in  `:` (line 14)
`C = EinsteinSum(A,id1,B,id2);`
Error in CDRX_mtex550 (line 66)
`tau_r_basal = abs(m_basal : S_macro_t)`

Here, `m_basal` is the velocity Gradient tensor defined as:
`m_basal = SchmidTensor(sS_basal);`
and `S_macro_t` is a stress tensor.

I also tried the alternative tensor contractions:
`tau_r = sS_basal.SchmidFactor(S_macro_t)`
or
`tau_r = EinsteinSum(m_basal,[-1 -2], S_macro_t,[-1 -2])`

but I got the same error  :(
Again, this was tested before in v5.4.0 and worked OK.

Thanks a lot. Regards.

**Reply by ralfHielscher** [reply]
could you please store your variables as a mat-file and attache it so that we can try to reproduce the error.

**Reply by MartinL75** [reply]
Yes, of course. Here's the m file and the variables.
[MartinL75.zip](https://github.com/mtex-toolbox/mtex/files/5628924/MartinL75.zip)

Thanks and regards.

Martín.-

**Comment by ralfHielscher** [comment]
Hi Martin,

we just released MTEX 5.5.1 which hopefully fixes these incompatibility issues with earlier Matlab versions. Could you please test it?

Thank you very much,

Ralf.

**Reply by MartinL75** [reply]
Hi Ralf,
Thanks a lot for the feedback.
I've just tested, but sadly I got the same error from above (tensor/EinsteinSum (line 86)... etc.).
Sorry I can't give further clues for helping to find a solution. I'll try to get a newer Matlab version.
Thanks again. Regards.

**Comment by ralfHielscher** [comment]
Hi Martin,

I'm very confident that I could identify the line of incompatibility. Please do the following:
```matlab
edit tensor/EinsteinSum.m
```
in line 76 you should see
``` matlab
  % TODO: later use M1 = M1 .* M2;
  %  M1 = bsxfun(@times, M1, M2);
  M1 = M1 .* M2;
```
replace this by
```matlab
  % TODO: later use M1 = M1 .* M2;
  M1 = bsxfun(@times, M1, M2);
  %M1 = M1 .* M2;
```
i.e., you need just to swap the comment sign between the last two lines.

This should do the trick.
Ralf.

**Reply by MartinL75** [reply]
Hi Ralf,
Sorry, but same error again :((

**Comment by ralfHielscher** [comment, marked as answer]
Could you please give MTEX 5.5.2 a last try to resolve this issue?

Ralf.

**Reply by MartinL75** [reply]
The same error showed up :(
I'm sorry

**Reply by MartinL75** [reply]
I'd like to point out that this error was solved when using the new v5.6.
Thanks a lot. Regards.
